File: app/src/main/java/com/example/data/remote/WhatsAppService.kt
package com.example.data.remote

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.util.Log
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.json.JSONArray
import org.json.JSONObject
import java.net.URL
import java.net.URLEncoder
import java.text.NumberFormat
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import javax.net.ssl.HttpsURLConnection

data class WhatsAppResult(
    val success: Boolean,
    val mode: String, // "API" or "DEEP_LINK"
    val message: String,
    val deepLinkUrl: String? = null
)

class WhatsAppService {

    private val TAG = "WhatsAppService"

    private val currencyFormat = NumberFormat.getCurrencyInstance(Locale("en", "IN")).apply {
        maximumFractionDigits = 0
    }

    private fun formatMoney(amount: Double): String {
        return currencyFormat.format(amount)
    }

    private fun formatDate(timestamp: Long): String {
        val sdf = SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH)
        return sdf.format(Date(timestamp))
    }

    fun cleanPhoneNumber(phone: String, defaultCountryCode: String = "91"): String {
        val digits = phone.filter { it.isDigit() }
        val clean10 = if (digits.length >= 10) digits.takeLast(10) else digits
        return "$defaultCountryCode$clean10"
    }

    fun buildBookingMessage(
        customerName: String,
        projectName: String,
        plotNumber: String,
        bookingAmount: Double,
        totalPrice: Double,
        bookingDate: Long = System.currentTimeMillis()
    ): String {
        return """
            Hello *$customerName*,

            Thank you for booking with us! We are delighted to confirm your plot booking. Here are your official booking details:

            📌 *Project Name:* $projectName
            🏡 *Plot Number:* Plot #$plotNumber
            💰 *Total Agreement Price:* ${formatMoney(totalPrice)}
            💳 *Booking Amount Paid:* ${formatMoney(bookingAmount)}
            📅 *Booking Date:* ${formatDate(bookingDate)}

            Your booking confirmation & payment receipt have been generated. Our operations team will reach out to you shortly with agreement documentation and payment schedule details.

            Thank you for trusting us with your dream property!

            Warm regards,
            Sales & Operations Team · Terra Plots
        """.trimIndent()
    }

    fun buildEmiStatementMessage(
        customerName: String,
        projectDetails: String,
        totalContractPrice: Double,
        totalAmountPaid: Double,
        remainingBalance: Double,
        installmentNumber: Int,
        totalInstallments: Int,
        nextDueDate: String,
        nextDueAmount: Double
    ): String {
        return """
            Dear *$customerName*,

            Here is your updated EMI payment statement and ledger summary for your property reservation:

            🏡 *Property:* $projectDetails
            💰 *Total Contract Price:* ${formatMoney(totalContractPrice)}
            ✅ *Total Amount Received:* ${formatMoney(totalAmountPaid)}
            ⏳ *Remaining Balance:* ${formatMoney(remainingBalance)}

            📈 *EMI Progress:*
            • Milestone: $installmentNumber of $totalInstallments
            • Next Due Date: $nextDueDate
            • Next Due Amount: ${formatMoney(nextDueAmount)}

            Payment receipts and ledger entries have been recorded in your customer account.
            For any queries, please reach out to our treasury desk.

            Best regards,
            Treasury & Accounts · Terra Plots
        """.trimIndent()
    }

    fun getWhatsAppDeepLink(phone: String, messageText: String): String {
        val cleanPhone = cleanPhoneNumber(phone)
        val encodedText = URLEncoder.encode(messageText, "UTF-8")
        return "https://wa.me/$cleanPhone?text=$encodedText"
    }

    fun launchWhatsAppIntent(context: Context, phone: String, messageText: String): Boolean {
        return try {
            val deepLink = getWhatsAppDeepLink(phone, messageText)
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(deepLink)).apply {
                flags = Intent.FLAG_ACTIVITY_NEW_TASK
            }
            context.startActivity(intent)
            true
        } catch (e: Exception) {
            Log.e(TAG, "Failed to launch WhatsApp intent: ${e.message}")
            false
        }
    }

    /**
     * Sends booking confirmation via Meta Cloud API using the plot_booking_confirmation template.
     */
    suspend fun sendBookingConfirmationApi(
        customerName: String,
        customerPhone: String,
        projectName: String,
        plotNumber: String,
        bookingAmount: Double,
        totalPrice: Double,
        bookingDate: Long = System.currentTimeMillis()
    ): WhatsAppResult = withContext(Dispatchers.IO) {
        val phoneWithCode = cleanPhoneNumber(customerPhone)
        val formattedMsg = buildBookingMessage(
            customerName = customerName,
            projectName = projectName,
            plotNumber = plotNumber,
            bookingAmount = bookingAmount,
            totalPrice = totalPrice,
            bookingDate = bookingDate
        )
        val deepLink = getWhatsAppDeepLink(customerPhone, formattedMsg)

        val token = AppConfig.WHATSAPP_API_TOKEN
        val phoneId = AppConfig.WHATSAPP_PHONE_NUMBER_ID
        val template = AppConfig.WHATSAPP_TEMPLATE_BOOKING

        if (token.isBlank() || phoneId.isBlank()) {
            return@withContext WhatsAppResult(
                success = false,
                mode = "DEEP_LINK",
                message = "Meta WhatsApp credentials not set. 1-tap WhatsApp ready.",
                deepLinkUrl = deepLink
            )
        }

        try {
            val endpoint = "https://graph.facebook.com/v19.0/$phoneId/messages"
            val url = URL(endpoint)
            val conn = url.openConnection() as HttpsURLConnection
            conn.requestMethod = "POST"
            conn.connectTimeout = 12000
            conn.readTimeout = 12000
            conn.doOutput = true
            conn.setRequestProperty("Authorization", "Bearer $token")
            conn.setRequestProperty("Content-Type", "application/json")

            // Meta Cloud API template body parameters
            val params = JSONArray().apply {
                put(JSONObject().apply { put("type", "text"); put("text", customerName) })
                put(JSONObject().apply { put("type", "text"); put("text", projectName) })
                put(JSONObject().apply { put("type", "text"); put("text", plotNumber) })
                put(JSONObject().apply { put("type", "text"); put("text", formatMoney(totalPrice)) })
                put(JSONObject().apply { put("type", "text"); put("text", formatMoney(bookingAmount)) })
                put(JSONObject().apply { put("type", "text"); put("text", formatDate(bookingDate)) })
            }

            val body = JSONObject().apply {
                put("messaging_product", "whatsapp")
                put("to", phoneWithCode)
                put("type", "template")
                put("template", JSONObject().apply {
                    put("name", template)
                    put("language", JSONObject().apply { put("code", "en") })
                    put("components", JSONArray().apply {
                        put(JSONObject().apply {
                            put("type", "body")
                            put("parameters", params)
                        })
                    })
                })
            }

            conn.outputStream.use { os ->
                os.write(body.toString().toByteArray(Charsets.UTF_8))
            }

            val code = conn.responseCode
            if (code in 200..299) {
                val resp = conn.inputStream.bufferedReader().use { it.readText() }
                Log.i(TAG, "Meta WhatsApp API Success: $resp")
                WhatsAppResult(
                    success = true,
                    mode = "API",
                    message = "Official WhatsApp confirmation sent to +$phoneWithCode!",
                    deepLinkUrl = deepLink
                )
            } else {
                val errResp = conn.errorStream?.bufferedReader()?.use { it.readText() } ?: "HTTP $code"
                Log.w(TAG, "Meta WhatsApp API Error ($code): $errResp")
                WhatsAppResult(
                    success = false,
                    mode = "API_FALLBACK",
                    message = "Meta API: HTTP $code. 1-tap WhatsApp dispatch ready.",
                    deepLinkUrl = deepLink
                )
            }
        } catch (e: Exception) {
            Log.e(TAG, "WhatsApp API exception", e)
            WhatsAppResult(
                success = false,
                mode = "DEEP_LINK",
                message = "Network dispatch: ${e.localizedMessage}. 1-tap WhatsApp link prepared.",
                deepLinkUrl = deepLink
            )
        }
    }
}

File: src/lib/whatsappService.ts
/**
 * WhatsApp Automation Service for Plot Perfect
 * Supports both Official Meta WhatsApp Cloud API / Provider webhooks
 * and 1-click WhatsApp web links for instant sales & ledger dispatch.
 */

export interface BookingWhatsAppPayload {
  customerName: string;
  customerPhone: string;
  projectName: string;
  plotNumber: string;
  totalPrice: number;
  bookingAmountPaid: number;
  bookingDate?: string;
  salesExecutiveName?: string;
}

export interface EMIStatementWhatsAppPayload {
  customerName: string;
  customerPhone: string;
  unitProjectDetails: string; // e.g. "Shree Durga Enclave - Plot #108"
  totalContractPrice: number;
  totalAmountRealized: number;
  remainingBalance: number;
  paidInstallmentsText: string; // e.g. "4 of 12 EMIs"
  pendingInstallmentsText: string; // e.g. "8 EMIs"
  nextDueDate?: string; // e.g. "10 Sep 2026"
  nextDueAmount: number;
  pdfDocumentUrl?: string;
  pdfFileName?: string;
}

const formatMoney = (val: number) =>
  `₹${Math.round(val || 0).toLocaleString("en-IN")}`;

const formatDate = (dateStr?: string) => {
  if (!dateStr) return new Date().toLocaleDateString("en-IN", { day: "numeric", month: "short", year: "numeric" });
  try {
    const [y, m, d] = dateStr.slice(0, 10).split("-").map(Number);
    return new Date(y, m - 1, d).toLocaleDateString("en-IN", { day: "numeric", month: "short", year: "numeric" });
  } catch {
    return dateStr;
  }
};

/**
 * Clean phone number to 10-digit format with country code (defaults to +91 India).
 */
export function formatWhatsAppPhone(phone: string, countryCode = "91"): string {
  const digits = phone.replace(/\D/g, "");
  const clean10 = digits.slice(-10);
  return `${countryCode}${clean10}`;
}

/**
 * Formats clean, professional text for WhatsApp booking confirmation message.
 */
export function buildBookingConfirmationMessage(payload: BookingWhatsAppPayload): string {
  const {
    customerName,
    projectName,
    plotNumber,
    totalPrice,
    bookingAmountPaid,
    bookingDate,
  } = payload;

  return `Hello *${customerName}*,

Thank you for booking with us! We are delighted to confirm your plot booking. Here are your official booking details:

📌 *Project Name:* ${projectName}
🏡 *Plot Number:* Plot #${plotNumber}
💰 *Total Agreement Price:* ${formatMoney(totalPrice)}
💳 *Booking Amount Paid:* ${formatMoney(bookingAmountPaid)}
📅 *Booking Date:* ${formatDate(bookingDate)}

Your booking confirmation & payment receipt have been generated. Our operations team will reach out to you shortly with agreement documentation and payment schedule details.

Thank you for trusting us with your dream property!

Warm regards,
Sales & Operations Team`;
}

/**
 * Formats clean, professional text for SBI Bank style EMI Statement WhatsApp message.
 */
export function buildEMIStatementMessage(payload: EMIStatementWhatsAppPayload): string {
  return `Dear *${payload.customerName}*,

Here is your updated EMI payment statement and ledger summary for your property reservation:

🏡 *Project & Plot:* ${payload.unitProjectDetails}
💰 *Total Contract Price:* ${formatMoney(payload.totalContractPrice)}
✅ *Total Amount Received:* ${formatMoney(payload.totalAmountRealized)}
⏳ *Remaining Balance:* ${formatMoney(payload.remainingBalance)}

📈 *EMI Progress:*
• Paid Installments: ${payload.paidInstallmentsText}
• Pending Installments: ${payload.pendingInstallmentsText}
📅 *Next EMI Due Date:* ${formatDate(payload.nextDueDate)}
💵 *Next Due Amount:* ${formatMoney(payload.nextDueAmount)}

Thank you for your prompt payments and continued trust with us!

Warm regards,
Accounts & Finance Team`;
}

/**
 * Generates direct 1-tap WhatsApp Web / Mobile deep link for Booking Confirmation.
 */
export function getWhatsAppDeepLink(payload: BookingWhatsAppPayload): string {
  const phoneWithCode = formatWhatsAppPhone(payload.customerPhone);
  const messageText = buildBookingConfirmationMessage(payload);
  return `https://wa.me/${phoneWithCode}?text=${encodeURIComponent(messageText)}`;
}

/**
 * Generates direct 1-tap WhatsApp Web / Mobile deep link for EMI Statement.
 */
export function getEMISatementWhatsAppDeepLink(payload: EMIStatementWhatsAppPayload): string {
  const phoneWithCode = formatWhatsAppPhone(payload.customerPhone);
  const messageText = buildEMIStatementMessage(payload);
  return `https://wa.me/${phoneWithCode}?text=${encodeURIComponent(messageText)}`;
}

const HARDCODED_ACCESS_TOKEN = "EAAONO18mvZBYBSSgXtZCJ9Ya6ZC0ZCZAuZBs1MoeP0zjrKLXLstjaIeeYRrZAtZBkEsiTXyAPQK17sI5bXfKwSlyucnVCu12q9bKvhhAydhEFn9HgpYb2JWtO5EbR9RBPQ1mWHiAgL4ZCNVD5sl90aHzZB7T5hW0kB1mO2EvQQZCxZA2g9a5nJEz0XD3Mp1p8TT5tBzX7xD7UXehIZAutmQYmUGVPVhksxHBaBhQZAZBaiZBQHZCjrsNhafT8ff6CZAzMK7oVbMJNDdzSc7pukDZCikbZBixUGjipl3N8QZDZD";
const HARDCODED_PHONE_NUMBER_ID = "1126770290524197";
const HARDCODED_BOOKING_TEMPLATE_NAME = "plot_booking_confirmation";
const HARDCODED_EMI_TEMPLATE_NAME = "customer_emi_statement_v2";

/**
 * Automated API Trigger for Booking Confirmation via Meta WhatsApp Cloud API.
 */
export async function sendBookingConfirmationWhatsApp(
  payload: BookingWhatsAppPayload,
  config?: {
    accessToken?: string;
    phoneNumberId?: string;
    templateName?: string;
  }
): Promise<{ success: boolean; mode: "api" | "deeplink"; message?: string; deepLink?: string }> {
  const accessToken = config?.accessToken || import.meta.env.VITE_WHATSAPP_API_TOKEN || HARDCODED_ACCESS_TOKEN;
  const phoneNumberId = config?.phoneNumberId || import.meta.env.VITE_WHATSAPP_PHONE_NUMBER_ID || HARDCODED_PHONE_NUMBER_ID;
  const templateName = config?.templateName || import.meta.env.VITE_WHATSAPP_TEMPLATE_NAME || HARDCODED_BOOKING_TEMPLATE_NAME;

  const deepLink = getWhatsAppDeepLink(payload);

  if (!accessToken || !phoneNumberId) {
    return {
      success: false,
      mode: "api",
      deepLink,
      message: "Meta WhatsApp API access token or Phone Number ID not configured.",
    };
  }

  const phoneWithCode = formatWhatsAppPhone(payload.customerPhone);
  const preferredLang = import.meta.env.VITE_WHATSAPP_LANGUAGE_CODE || "en";

  async function postTemplateWithParams(langCode: string, params: any[]) {
    return fetch(
      `https://graph.facebook.com/v19.0/${phoneNumberId}/messages`,
      {
        method: "POST",
        headers: {
          Authorization: `Bearer ${accessToken}`,
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          messaging_product: "whatsapp",
          to: phoneWithCode,
          type: "template",
          template: {
            name: templateName,
            language: { code: langCode },
            components: [
              {
                type: "body",
                parameters: params,
              },
            ],
          },
        }),
      }
    );
  }

  // 6 Parameters expected by plot_booking_confirmation template
  const params6 = [
    { type: "text", text: payload.customerName },
    { type: "text", text: payload.projectName },
    { type: "text", text: String(payload.plotNumber) },
    { type: "text", text: formatMoney(payload.totalPrice) },
    { type: "text", text: formatMoney(payload.bookingAmountPaid) },
    { type: "text", text: formatDate(payload.bookingDate) },
  ];

  // 4 Parameters fallback
  const params4 = [
    { type: "text", text: payload.customerName },
    { type: "text", text: `${payload.projectName} · Plot #${payload.plotNumber}` },
    { type: "text", text: formatMoney(payload.bookingAmountPaid) },
    { type: "text", text: formatMoney(payload.totalPrice) },
  ];

  try {
    // 1. Try 6-parameter template first (matches Meta template spec)
    let response = await postTemplateWithParams(preferredLang, params6);
    let result = await response.json();

    // 2. If parameter count mismatch (#132000), try 4-parameter layout
    if (!response.ok && (result.error?.code === 132000 || result.error?.message?.includes("does not match"))) {
      response = await postTemplateWithParams(preferredLang, params4);
      result = await response.json();
    }

    // 3. If language code mismatch (#132001), try en_US
    if (!response.ok && (result.error?.code === 132001 || result.error?.message?.includes("translation"))) {
      const alternateLang = preferredLang === "en" ? "en_US" : "en";
      response = await postTemplateWithParams(alternateLang, params6);
      result = await response.json();
    }

    if (!response.ok) {
      console.warn("[Meta WhatsApp Cloud API Error]", result);
      const errCode = result.error?.code || "API";
      const errMsg = result.error?.error_data?.details || result.error?.message || "Meta Cloud API call failed";

      let userMsg = `Meta API Error (#${errCode}): ${errMsg}`;
      if (errCode === 131005 || errCode === 190) {
        userMsg = `Meta Access Token Expired (#${errCode}). Please refresh VITE_WHATSAPP_API_TOKEN in .env file!`;
      }

      return {
        success: false,
        mode: "api",
        deepLink,
        message: userMsg,
      };
    }

    return {
      success: true,
      mode: "api",
      message: `✅ Booking confirmation message successfully dispatched via Meta WhatsApp API to +${phoneWithCode}!`,
    };
  } catch (err: any) {
    return {
      success: false,
      mode: "api",
      deepLink,
      message: `Meta API Dispatch Error: ${err.message || "Network request failed"}`,
    };
  }
}

/**
 * Automated API Trigger for SBI-Style EMI Statement WhatsApp Dispatch.
 */
export async function sendEMIStatementWhatsApp(
  payload: EMIStatementWhatsAppPayload,
  config?: {
    accessToken?: string;
    phoneNumberId?: string;
    templateName?: string;
  }
): Promise<{ success: boolean; mode: "api" | "deeplink"; message?: string; deepLink?: string }> {
  const accessToken = config?.accessToken || import.meta.env.VITE_WHATSAPP_API_TOKEN || HARDCODED_ACCESS_TOKEN;
  const phoneNumberId = config?.phoneNumberId || import.meta.env.VITE_WHATSAPP_PHONE_NUMBER_ID || HARDCODED_PHONE_NUMBER_ID;
  const templateName = config?.templateName || import.meta.env.VITE_WHATSAPP_EMI_TEMPLATE_NAME || HARDCODED_EMI_TEMPLATE_NAME;

  const deepLink = getEMISatementWhatsAppDeepLink(payload);

  if (!accessToken || !phoneNumberId) {
    return {
      success: true,
      mode: "deeplink",
      deepLink,
      message: "WhatsApp API credentials not set. Generated 1-tap WhatsApp link.",
    };
  }

  const phoneWithCode = formatWhatsAppPhone(payload.customerPhone);
  const preferredLang = import.meta.env.VITE_WHATSAPP_LANGUAGE_CODE || "en";

  async function postTemplate(langCode: string, headerMode: "document" | "text" | "none" = "document") {
    const components: any[] = [];
    if (headerMode === "document" && payload.pdfDocumentUrl) {
      components.push({
        type: "header",
        parameters: [
          {
            type: "document",
            document: {
              link: payload.pdfDocumentUrl,
              filename: payload.pdfFileName || "EMI_Statement.pdf",
            },
          },
        ],
      });
    } else if (headerMode === "text") {
      components.push({
        type: "header",
        parameters: [
          {
            type: "text",
            text: "HAEGL Tech",
          },
        ],
      });
    }

    components.push({
      type: "body",
      parameters: [
        { type: "text", text: payload.customerName },                // {{1}}
        { type: "text", text: payload.unitProjectDetails },          // {{2}}
        { type: "text", text: formatMoney(payload.totalContractPrice) }, // {{3}}
        { type: "text", text: formatMoney(payload.totalAmountRealized) }, // {{4}}
        { type: "text", text: formatMoney(payload.remainingBalance) },   // {{5}}
        { type: "text", text: payload.paidInstallmentsText },        // {{6}}
        { type: "text", text: payload.pendingInstallmentsText },     // {{7}}
        { type: "text", text: formatDate(payload.nextDueDate) },     // {{8}}
        { type: "text", text: formatMoney(payload.nextDueAmount) },  // {{9}}
      ],
    });

    return fetch(
      `https://graph.facebook.com/v19.0/${phoneNumberId}/messages`,
      {
        method: "POST",
        headers: {
          Authorization: `Bearer ${accessToken}`,
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          messaging_product: "whatsapp",
          to: phoneWithCode,
          type: "template",
          template: {
            name: templateName,
            language: { code: langCode },
            components,
          },
        }),
      }
    );
  }

  async function postDirectDocument() {
    if (!payload.pdfDocumentUrl) return null;
    try {
      const res = await fetch(
        `https://graph.facebook.com/v19.0/${phoneNumberId}/messages`,
        {
          method: "POST",
          headers: {
            Authorization: `Bearer ${accessToken}`,
            "Content-Type": "application/json",
          },
          body: JSON.stringify({
            messaging_product: "whatsapp",
            to: phoneWithCode,
            type: "document",
            document: {
              link: payload.pdfDocumentUrl,
              filename: payload.pdfFileName || "EMI_Statement.pdf",
              caption: `📄 Official EMI Statement PDF for ${payload.customerName} (${payload.unitProjectDetails})\n💰 Contract Value: ${formatMoney(payload.totalContractPrice)}\n✅ Received: ${formatMoney(payload.totalAmountRealized)}\n⏳ Remaining Balance: ${formatMoney(payload.remainingBalance)}`,
            },
          }),
        }
      );
      const data = await res.json();
      console.log("[WhatsApp Direct PDF Document API Response]", data);
      return res.ok;
    } catch (e) {
      console.warn("[WhatsApp Direct PDF Error]", e);
      return false;
    }
  }

  try {
    // 1. Try single Template Message with Document Header if PDF link exists
    if (payload.pdfDocumentUrl) {
      let docTemplateResp = await postTemplate(preferredLang, "document");
      let docTemplateResult = await docTemplateResp.json();

      if (!docTemplateResp.ok && (docTemplateResult.error?.code === 132001 || docTemplateResult.error?.message?.includes("translation"))) {
        const alternateLang = preferredLang === "en" ? "en_US" : "en";
        docTemplateResp = await postTemplate(alternateLang, "document");
        docTemplateResult = await docTemplateResp.json();
      }

      // If Meta approved document header on template, single message succeeded!
      if (docTemplateResp.ok) {
        return {
          success: true,
          mode: "api",
          message: `📄 Single PDF Statement message successfully sent to ${phoneWithCode} via WhatsApp!`,
        };
      }

      // If Meta returned #132012 (Header format mismatch, expected TEXT), send 1 single Direct Document Message with full caption!
      if (docTemplateResult.error?.code === 132012 || docTemplateResult.error?.message?.includes("Format mismatch") || docTemplateResult.error?.message?.includes("header")) {
        console.log("[WhatsApp EMI API] Template header mismatch. Dispatching 1 single Direct PDF Document Message...");
        const directOk = await postDirectDocument();
        if (directOk) {
          return {
            success: true,
            mode: "api",
            message: `📄 Single PDF Statement document successfully sent to ${phoneWithCode} via WhatsApp!`,
          };
        }
      }
    }

    // 2. Standard single Template Message dispatch (Text only)
    let response = await postTemplate(preferredLang, "none");
    let result = await response.json();

    if (!response.ok && (result.error?.code === 132001 || result.error?.message?.includes("translation"))) {
      const alternateLang = preferredLang === "en" ? "en_US" : "en";
      response = await postTemplate(alternateLang, "none");
      result = await response.json();
    }

    if (!response.ok) {
      console.warn("[WhatsApp EMI API Error]", result);
      const errMsg = result.error?.error_data?.details || result.error?.message || "WhatsApp API call failed";
      return {
        success: false,
        mode: "api",
        deepLink,
        message: `(#${result.error?.code || "API"}) ${errMsg}`,
      };
    }

    return {
      success: true,
      mode: "api",
      message: `EMI Statement summary successfully sent to ${phoneWithCode} via WhatsApp!`,
    };
  } catch (err: any) {
    return {
      success: false,
      mode: "deeplink",
      deepLink,
      message: err.message || "Failed to dispatch WhatsApp statement via API",
    };
  }
}
